src/auxiliary/auxiliary.jl

```diff
@@ -370,4 +370,14 @@ function download(src_url, file_path)
 
     return file_path
 end
+
+abstract type MeshFile{NDIMS} end
+
+struct GmshFile{NDIMS} <: MeshFile{NDIMS}
+    path::String
+end
+
+struct AbaqusFile{NDIMS} <: MeshFile{NDIMS}
+    path::String
+end
```

Comment:
If they should remain public (due to reasons), a docstring would be helpful (plus, probably, a convenience ctor that allows one to write `AbaqusFile(ndims, path)` instead).

Reply:
I move them to `t8code_mesh.jl` for internal use.